### PR TITLE
Prevent infinite looping from hash table probe operations

### PR DESCRIPTION
**Pull request by xiaoxmeng**
There are infinite loops in hash table probe operations that
break until find the search entry or find out an empty slot.
This could result in infinite loop when there are some bugs in
the relevant code path. To prevent this, we shall break out of
the loop if we have searched through all the buckets in the
table

### CODE REVIEW

**Review by mbasmanova — APPROVED** [review]
@xiaoxmeng Thanks. Maybe update PR description to explain what this change is about. The title alone might be misleading.

File: velox/exec/HashTable.cpp [hunk 1]
```diff
@@ -249,6 +250,8 @@ class ProbeState {
           reinterpret_cast<uint8_t*>(table.table_), bucketOffset_);
       hits_ = simd::toBitMask(tagsInTable_ == wantedTags_) & kFullMask;
     }
+    // Throws here if we have looped through all the buckets in the table.
+    VELOX_FAIL("Have looped through all the buckets in table");
```
Comment (xumingming) [inline velox/exec/HashTable.cpp:254]: It should fail here because there must be some empty slots so not all buckets should be probed?

Comment (xiaoxmeng) [reply velox/exec/HashTable.cpp:254]: We only reach here if we couldn't find an empty slot. This is sort of preventing any bugs in the relevant code path that could lead to infinite loop.

File: velox/exec/HashTable.cpp [hunk 2]
```diff
@@ -229,7 +229,8 @@ class ProbeState {
       return group_;
     }
     const auto kEmptyGroup = BaseHashTable::TagVector::broadcast(kEmptyTag);
-    for (;;) {
+    for (int64_t numProbedBuckets = 0; numProbedBuckets < table.numBuckets();
+         ++numProbedBuckets) {
```
Comment (skadilover) [inline velox/exec/HashTable.cpp:233]: @xiaoxmeng When searching in a bucket, continue will increase numProbedBuckets. Is this behavior expected?

```
        loadNextHit<Operation::kProbe>(
            table, -static_cast<int32_t>(sizeof(normalized_key_t)));
        if (RowContainer::normalizedKey(group_) == keys[row_]) {
          table.incrementHits();
          return group_;
        }
        continue;
```

**Pull request closed without merging**